### PR TITLE
Generics task

### PR DESCRIPTION
**Pull request by Veboro**

### CODE REVIEW

File: src/Num.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        return "In the List we have " + cout + " numbers";
+
+    }
+
+    //method which takes two lists and index and replaces element
+    // at given position in the first collection with corresponding
+    // element from the second one, without possibility to do vice versa
+
+    public void replaceList(List<T> list1, List<T> list2, int index){
+        list1.set(index, list2.get(index));
```
Comment (stefankiv) [inline src/Num.java]: 4. Create utility method, which takes two lists and index and replaces element at given position in the first collection with corresponding element from the second one, without possibility to do vice versa.

It shouldn't be possible to write `list2.set(index, list1.get(index));`. Can you try it?

File: src/Num.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    public void switchList(List<T> list1, List<T> list2, int index){
+        T tmp = list1.get(index);
+        list1.set(index, list2.get(index));
+        list2.set(index, tmp);
+    }
+
+    //utility method, which finds maximum element in the given list.
+    public Comparable maximumElement(List list){
+        return Collections.max(list);
+    }
```
Comment (stefankiv) [inline src/Num.java]: Can you try this code:
`class Person {}`
`new Num<>(Collections.empyList()).maximumElement(Arrays.asList(new Person(), new Person()))`

File: src/Num.java (hunk 1 above)
Comment (Veboro) [reply src/Num.java]: And if so

    public void replaceList(List<? super T> list1, List<? extends T> list2, int index){
        list1.set(index, list2.get(index));
    }

File: src/Num.java (hunk 2 above)
Comment (Veboro) [reply src/Num.java]: I slightly corrected Person 

        class Person implements Comparable {

            @Override
            public int compareTo(Object o) {
                return 0;
            }
        }

Now all work

Comment (Veboro) [reply src/Num.java]: or 

    public <T extends Comparable<? super T>> T maximumElement(Collection<T> list){
        return Collections.max(list);
    }

**Review by stefankiv — APPROVED** [review]

**Pull request closed without merging**